internal/services/privatednsresolver/private_dns_resolver_forwarding_rule_data_source.go

```diff
@@ -0,0 +1,175 @@
+	DomainName             string                           `tfschema:"domain_name"`
+	Enabled                bool                             `tfschema:"enabled"`
+	Metadata               map[string]string                `tfschema:"metadata"`
+	TargetDnsServers       []TargetDnsServerDataSourceModel `tfschema:"target_dns_servers"`
+}
+
+type TargetDnsServerDataSourceModel struct {
+	IPAddress string `tfschema:"ip_address"`
+	Port      int64  `tfschema:"port"`
+}
+
+type PrivateDNSResolverForwardingRuleDataSource struct{}
+
+var _ sdk.DataSource = PrivateDNSResolverForwardingRuleDataSource{}
+
+func (r PrivateDNSResolverForwardingRuleDataSource) ResourceType() string {
+	return "azurerm_private_dns_resolver_forwarding_rule"
+}
+
+func (r PrivateDNSResolverForwardingRuleDataSource) ModelObject() interface{} {
+	return &PrivateDNSResolverForwardingRuleModel{}
+}
+
+func (r PrivateDNSResolverForwardingRuleDataSource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return forwardingrules.ValidateForwardingRuleID
+}
+
+func (r PrivateDNSResolverForwardingRuleDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"dns_forwarding_ruleset_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: dnsforwardingrulesets.ValidateDnsForwardingRulesetID,
+		},
+	}
+}
+
+func (r PrivateDNSResolverForwardingRuleDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"domain_name": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"target_dns_servers": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"ip_address": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"port": {
+						Type:     pluginsdk.TypeInt,
+						Computed: true,
+					},
+				},
+			},
+		},
+
+		"enabled": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+
+		"metadata": {
+			Type:     pluginsdk.TypeMap,
+			Computed: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+	}
+}
+
+func (r PrivateDNSResolverForwardingRuleDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.PrivateDnsResolver.ForwardingRulesClient
+
+			var state PrivateDNSResolverForwardingRuleDataSourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			dnsForwardingRulesetId, err := forwardingrules.ParseDnsForwardingRulesetID(state.DnsForwardingRulesetId)
+			if err != nil {
+				return err
+			}
+
+			var top int64 = 1
+			resp, err := client.ListCompleteMatchingPredicate(ctx, *dnsForwardingRulesetId,
```

Comment:
Similar as in the last PR, we should be able to get this resource directly rather than looping through a list.

Reply:
done